### PR TITLE
HTTPS-2 : TLS/SSL support added; Threading development ready

### PR DESCRIPTION
**Pull request by 0xvpr**
Reverted to using libhttpserver & libmicrohttpd for the project.

- Started using an experimental  "ServerOptions" struct for the environment variables
- Hard coded certs and keys

### CODE REVIEW

File: Server/Tests/TestServerOptions.hpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    assert(so.InternetProtocol() == IPV4_PROTOCOL);
+    assert(so.Timeout() == 60);
+    assert(so.MaxConnections() == 255);
+    assert(so.MaxThreads() == 8);
+    assert(so.ThreadCount() == 4); // TODO: Change to $n_proc
+    assert(so.MemoryLimit() == 32768);
+
+}
+
+void test_server_options_set_environment() {
```
Comment (calvindo95) [inline Server/Tests/TestServerOptions.hpp]: It looks like this test is still comparing to default options. Need to check against non-default values

File: Server/Includes/Data.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        "BBYEFJcUvpjvE5fF/yzUshkWDpdYiQh/MAsGCSqGSIb3DQEBBQOCAQEARP7eKSB2\n"
+        "RNd6XjEjK0SrxtoTnxS3nw9sfcS7/qD1+XHdObtDFqGNSjGYFB3Gpx8fpQhCXdoN\n"
+        "8QUs3/5ZVa5yjZMQewWBgz8kNbnbH40F2y81MHITxxCe1Y+qqHWwVaYLsiOTqj2/\n"
+        "0S3QjEJ9tvklmg7JX09HC4m5QRYfWBeQLD1u8ZjA1Sf1xJriomFVyRLI2VPO2bNe\n"
+        "JDMXWuP+8kMC7gEvUnJ7A92Y2yrhu3QI3bjPk8uSpHea19Q77tul1UVBJ5g+zpH3\n"
+        "OsF5p0MyaVf09GTzcLds5nE/osTdXGUyHJapWReVmPm3Zn6gqYlnzD99z+DPIgIV\n"
+        "RhZvQx74NQnS6g==\n"
+        "-----END CERTIFICATE-----\n";
+
+    constexpr const char* const raw_mem_key =
```
Comment (calvindo95) [inline Server/Includes/Data.hpp:24]: The key should be private, otherwise I could make my own certificate with it.

Comment (0xvpr) [reply Server/Includes/Data.hpp:24]: I should probably add a note to look at Main.cpp somewhere in there or explain that its only used if a private key / public crt wasn't found.

Comment (0xvpr) [reply Server/Includes/Data.hpp:24]:
```cpp
    // Set server crt
    if (std::filesystem::is_regular_file(so.MemCert())) {
        cw.https_mem_cert(so.MemCert());
    } else {
        cw.raw_https_mem_cert(data::raw_mem_cert);
    }

    // Set server key
    if (std::filesystem::is_regular_file(so.MemKey())) {
        cw.https_mem_key(so.MemKey());
    } else {
        cw.raw_https_mem_key(data::raw_mem_key);
    }
```

File: Server/Tests/TestServerOptions.hpp (hunk 1 above)
Comment (0xvpr) [reply Server/Tests/TestServerOptions.hpp]: Good catch.

**Pull request merged**